pokemongo_bot/cell_workers/evolve_pokemon.py

```diff
@@ -0,0 +1,152 @@
+from pokemongo_bot import logger
+from pokemongo_bot.human_behaviour import sleep
+from pokemongo_bot.item_list import Item
+from pokemongo_bot.cell_workers.base_task import BaseTask
+
+class EvolvePokemon(BaseTask):
+    def initialize(self):
+        self.api = self.bot.api
+        self.evolve_all = self.config.get('evolve_all', [])
+        self.evolve_speed = self.config.get('evolve_speed', 3.7)
+        self.first_evolve_by = self.config.get('first_evolve_by', 'cp')
+        self.evolve_above_cp = self.config.get('evolve_above_cp', 500)
+        self.evolve_above_iv = self.config.get('evolve_above_iv', 0.8)
+        self.cp_iv_logic = self.config.get('logic', 'or')
+        self.use_lucky_egg = self.config.get('use_lucky_egg', False)
+        self._validate_config()
+
+    def _validate_config(self):
+        if isinstance(self.evolve_all, basestring):
+            self.evolve_all = [str(pokemon_name) for pokemon_name in self.evolve_all.split(',')]
+
+    def work(self):
+        if not self._should_run():
+            return
+
+        response_dict = self.api.get_inventory()
+        inventory_items = response_dict.get('responses', {}).get('GET_INVENTORY', {}).get('inventory_delta', {}).get(
```

Comment:
@douglascamata 
If any of those key is not exist,  inventory_items will get an valid empty dict. Then it will be passed to the subsequence methods.